libp2p/src/builder/phase/swarm.rs

```diff
@@ -0,0 +1,80 @@
+use super::*;
+use crate::SwarmBuilder;
+use libp2p_swarm::{NetworkBehaviour, Swarm};
+use std::marker::PhantomData;
+
+pub struct SwarmPhase<T, B> {
+    pub(crate) behaviour: B,
+    pub(crate) transport: T,
+}
+
+impl<T, B, Provider> SwarmBuilder<Provider, SwarmPhase<T, B>> {
+    pub fn with_swarm_config(
```

Comment:
I don't feel strongly but do we need the `swarm` mention here? This function is defined on a struct called `SwarmBuilder`. Isn't that enough context?

Reply:
With all the other type above, e.g. tcp, quic, relay, behaviour, I find it more intuitive to stress that this is the config of the swarm now. Thus I prefer `with_swarm_config` instead of `with_config`.